gateway/src/proxy.rs

```diff
@@ -41,6 +44,7 @@ pub struct ProxyState {
     gateway: Arc<GatewayService>,
     task_sender: Sender<BoxedTask>,
     public: FQDN,
+    project_cache: Mutex<TimedCache<ProjectName, FindProjectPayload>>,
```

Comment:
We have [a cache](https://github.com/shuttle-hq/shuttle/blob/main/common/src/backends/cache.rs) implementation in common. We use it in the gateway with an rw lock, which makes sense for the jwt, but perhaps not for this short-lived cache. It would be worth testing, though. Should we use the existing cache here rather than this crate, or does this crate offer any advantages, e.g. increased performance?

Reply:
The TTL cache supports an individual duration per item, which we don't need, but it does looks more lightweight. I did try a RwLock instead of Mutex, but the TimedCache has a hit/miss counter, which means that `get_cached()` is mutable, and needs write access.
Regardless, I don't think it should matter much in terms of performance.

Reply:
IMO it's worth checking, given that this is a performance optimization, and you're also adding a new dependency by not using the existing cache.

Reply:
No meaningful difference, but I'll use the existing one since it is lighter and no new dep.

Reply:
Thanks for checking!